rxkeystore/src/androidTest/java/com/nexenio/rxkeystore/provider/cipher/asymmetric/rsa/RsaCipherProviderTest.java

```diff
@@ -51,11 +59,73 @@ public void generateSecretKey_matchingKeyPairs_sameSecretKey() {
     public void encrypt_invalidDataLength_emitsError() {
         byte[] unencryptedBytes = LOREM_IPSUM_LONG.getBytes(StandardCharsets.UTF_8);
 
-        asymmetricCryptoProvider.generateKeyPair(ALIAS_NEW, context)
-                .flatMap(keyPair -> asymmetricCryptoProvider.encrypt(unencryptedBytes, keyPair.getPublic())
-                        .flatMap(encryptedBytesAndIV -> asymmetricCryptoProvider.decrypt(encryptedBytesAndIV.first, encryptedBytesAndIV.second, keyPair.getPrivate())))
+        bouncyCastleProvider.generateKeyPair(ALIAS_NEW, context)
+                .flatMap(keyPair -> bouncyCastleProvider.encrypt(unencryptedBytes, keyPair.getPublic())
+                        .flatMap(encryptedBytesAndIV -> bouncyCastleProvider.decrypt(encryptedBytesAndIV.first, encryptedBytesAndIV.second, keyPair.getPrivate())))
                 .test()
                 .assertError(RxEncryptionException.class);
     }
 
-}
\ No newline at end of file
+    @Test
+    public void encrypt_publicKey$wrongPurpose$defaultProvider_emitsNoError() {
+        byte[] unencryptedBytes = Base64.decode(ENCODED_SYMMETRIC_KEY, Base64.DEFAULT);
+
+        defaultProvider.generateKeyPair(ALIAS_NEW, KeyProperties.PURPOSE_DECRYPT, context)
+                .flatMap(keyPair -> defaultProvider.encrypt(unencryptedBytes, keyPair.getPublic()))
+                .test()
+                .assertNoErrors();
+    }
+
+    @Test
+    public void encrypt_privateKey$wrongPurpose$defaultProvider_emitsError() {
+        byte[] unencryptedBytes = Base64.decode(ENCODED_SYMMETRIC_KEY, Base64.DEFAULT);
+
+        defaultProvider.generateKeyPair(ALIAS_NEW, KeyProperties.PURPOSE_DECRYPT, context)
+                .flatMap(keyPair -> defaultProvider.encrypt(unencryptedBytes, keyPair.getPrivate()))
+                .test()
+                .assertError(RxEncryptionException.class);
+    }
+
+    @Test
+    public void encrypt_public$wrongPurpose$bouncyCastleProvider_emitsNoError() {
```

Comment:
I added this test to show this behavior. In my eyes it is important to know that it was no effect on some key stores, because according to the documentation of the key purposes all stores should fail for mismatching purposes.